### PR TITLE
feat: Add option to omit empty versions from output

### PR DESCRIPTION
**Pull request by svenax**
Ad option to omit versions with no sections from output.

### CODE REVIEW

**Comment by pawamoy** [comment]
Thanks! Can you keep the changes atomic though?

One PR for the feature, one for the code blocks languages, and one for the Atom commit convention? By the way, what is the reasoning behind this last one? This will either break the Atom convention or just make it similar to the Conventional one.

**Comment by pawamoy** [comment]
I see, thanks. I'd like that we fix this in another PR :slightly_smiling_face:

**Comment by svenax** [comment]
Sure. If I revert that change, `mypy` will fail like this.

```
src/git_changelog/build.py:160: error: Only concrete class can be given where "tuple[str, type[CommitConvention]]" is expected  [type-abstract]
```

To get around it, that line needs

```python
 "atom": AtomConvention,  # type: ignore[type-abstract]
```

Should I add that?

**Comment by pawamoy** [comment]
Sure, fine to ignore to type issue in this PR. Thanks!

I'm wondering if we could even just drop this Atom convention... The Atom repository is archived, I can't find any docs about this commit convention on the web. We could deprecate it first of course.

**Comment by svenax** [comment]
I split the changes into separate commits. I would be fine with removing Atom convention. As of now, does it even do anything?

**Comment by pawamoy** [comment]
Can you change the Mypy commit type to `chore` or `ci`? `fix` is reserved for actual bug fixes.

I just tried using the Atom convention: it's broken, so let's remove it. I'll handle it once this PR is merged.

File: src/git_changelog/cli.py [hunk 1]
```diff
@@ -226,6 +226,14 @@ def get_parser() -> argparse.ArgumentParser:
         dest="parse_trailers",
         help="Parse Git trailers in the commit message. See https://git-scm.com/docs/git-interpret-trailers. Default: %(default)s.",
     )
+    parser.add_argument(
+        "--omit-empty",
+        "--omit-empty-versions",
```
Comment (pawamoy) [inline src/git_changelog/cli.py:231]: Let's keep only one long option, and add a short one:

```suggestion
        "--E",
        "--omit-empty-versions",
```

Comment (svenax) [reply src/git_changelog/cli.py]: Done

**Comment by pawamoy** [comment]
Heh, sorry, can you squash your fix commit into the feat one :sweat_smile:? It's not a fix either, more like a refactor :stuck_out_tongue: 

If you contribute again in the future I recommend splitting different changes into different PRs, that makes it easier for me to squash and reword commits if needed :slightly_smiling_face: (obviously I could checkout your branch, rebase myself and force-push but that's not very mindful).

**Review by pawamoy — APPROVED** [review]
Awesome, thanks a lot!

**Pull request merged**